mne/io/kit/tests/test_kit.py

```diff
@@ -185,4 +186,24 @@ def test_hsp_elp():
     assert_array_almost_equal(pts_elp_in_dev, pts_txt_in_dev, decimal=5)
 
 
+def test_decimate():
+    """Test decimation of digitizer headshapes with too many points. """
+    # create fake head points from sphere, and scale to similar numbers as hsp
+    hsp = _get_ico_surface(5)['rr']
+    hsp /= 10
+
+    # read in raw data using spherical hsp, and extract new hsp
+    raw = read_raw_kit(sqd_path, mrk_path, elp_txt_path, hsp)
```

Comment:
I wonder if it would be worth writing the points to a temporary `.txt` file too, since the actual unit conversion happens in `_read_dig_points()` which is not called when the points are specified as array.